64-bit target with `nightly` feature doesn't compile anymore.

Since #300, using `nightly` feature with a 64-bit target doesn't compile anymore...

```
   Compiling pagecache v0.4.3 (file:///home/rsmet/src/sled/crates/pagecache)
error[E0308]: mismatched types
    --> crates/pagecache/src/io/segment.rs:1212:18
     |
1212 |         max_lsn: std::i64::MAX,
     |                  ^^^^^^^^^^^^^ expected isize, found i64

error: aborting due to previous error
```

--- a/crates/pagecache/src/io/segment.rs
+++ b/crates/pagecache/src/io/segment.rs
@@ -1208,9 +1208,9 @@ pub fn raw_segment_iter_from(
 
     Ok(LogIter {
         config: config.clone(),
-        #[cfg(feature = "nightly")]
+        #[cfg(target_pointer_width = "32")]
         max_lsn: std::i64::MAX,
-        #[cfg(not(feature = "nightly"))]
+        #[cfg(target_pointer_width = "64")]
         max_lsn: std::isize::MAX,
         cur_lsn: SEG_HEADER_LEN as Lsn,
         segment_base: None,

